citations-impl/impl/src/java/org/sakaiproject/citation/impl/DbCitationService.java

```diff
@@ -1377,6 +1734,42 @@ public Object readSqlResultRecord(ResultSet result)
 
 	}
 
+	public class CitationCollectionOrderReader implements SqlReader
+	{
+
+		/* (non-Javadoc)
+         * @see org.sakaiproject.db.api.SqlReader#readSqlResultRecord(java.sql.ResultSet)
+         */
+		public Object readSqlResultRecord(ResultSet result)
+		{
+			CitationCollectionOrder citationCollectionOrder = null;
+
+			String collectionId = null;
+			String citationId = null;
+			int location = 0;
+			CitationCollectionOrder.SectionType sectionType = null;
+			String value = null;
+			try
+			{
+				collectionId = result.getString(1);
+				citationId = result.getString(2);
+				location = result.getInt(3);
+				sectionType = CitationCollectionOrder.SectionType.valueOf(result.getString(4));
+				value = result.getString(5);
+
+				citationCollectionOrder = new CitationCollectionOrder(collectionId, citationId, location, sectionType, value);
+			}
+			catch (SQLException e)
+			{
+				M_log.debug("TripleReader: problem reading CitationCollectionOrder from result: collectionId(" + collectionId + ") location(" + location
```

Comment:
I'd log this as a warning as it's something we want to know about during production and we don't ever expect it to happen.